EIPS/eip-7775.md

```diff
@@ -0,0 +1,119 @@
+---
+eip: 7775
+title: BURN opcode
+description: An opcode to burn native ether at the given address
+author: Dev Bear (@itsdevbear)
+discussions-to: https://ethereum-magicians.org/t/eip-7775-burn-opcode/21287
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-09-30
+---
+
+## Abstract
+
+This proposal introduces a `BURN` opcode to the EVM. When called, the opcode is to burn native ether at the address of the current evm context.
+
+## Motivation
+
+The motivation for this proposal is to provide a standardized and efficient way to burn native ether directly within the EVM. Historically, contracts such as the BeaconDepositContract have "burned" ether by making it irrecoverable from the given address. This approach can lead to confusion and potential misuse. By introducing a dedicated `BURN` opcode, we can ensure a clear and consistent method for burning native ether. This could become useful for Ethereum L2s when transferring ether back to the L1, as well as other EVM L1 chains that could leverage this for their cryptoeconomics.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Behaviour
+
+The `BURN` opcode `(0xFC)` is introduced with the following behavior:
+
+1. Pops one 32-byte word from the top of the stack, treating these bytes as the `uint256` amount of native ether to be burned.
+2. Retrieves the current address from the EVM execution context.
+3. Checks the balance of the current address.
+4. If the amount to be burned is greater than the balance of the current address, the opcode `MUST` revert.
+5. If the amount to be burned is 0, the execution `MUST NOT` revert.
+6. Subtracts the amount from the current address's native ether balance.
+
+When `BURN` is used in the context of a DELEGATECALL or CALLCODE, the contract whose balance is to be manipulated is the contract that issued the DELEGATECALL or CALLCODE instruction.
+
+When `BURN` is used in the context of a STATICCALL the call `MUST` revert.
+
+### Gas Cost
+
+The base gas cost for the `BURN` opcode is 100 gas. The dynamic gas cost is determined as follows:
+
+1. If the value to be burned is 0, the dynamic gas cost is 0.
```

Comment:
0 is also too low, also for a no-op. A no-op `JUMPDEST` costs 1 gas. Also note we have to pop one item from stack here. `3` would be reasonable here (same as the `EQ` operator, which pops two items and compares those)

Reply:
This is just the dynamic cost. The min cost of the code is always 100

Reply:
@jochem-brouwer thoughts?

Reply:
The base cost is too low, see: https://github.com/ethereum/EIPs/pull/8914#discussion_r1798660732

Reply:
Or wait, maybe dynamic should be jumped if the target is cold. But the cost is too low for cold accounts.

Reply:
@jochem-brouwer I think the dyn can be 0 if account is cold, since we can short circuit on balance. Updated to reflect. LMK what u think.